moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp

```diff
@@ -119,7 +119,8 @@ bool SynchronizedStringParameter::shouldPublish()
 
 bool SynchronizedStringParameter::waitForMessage(const rclcpp::Duration timeout)
 {
-  string_subscriber_ = node_->create_subscription<std_msgs::msg::String>(
+  auto temp_node = std::make_shared<rclcpp::Node>("synchronized_string_parameter");
+  string_subscriber_ = temp_node->create_subscription<std_msgs::msg::String>(
```

Comment:
Two things, this should probably be a const variable.  Secondly, I am worried about name collisions with that name.  I tried to find documentation on if there is some way to get a unique random name in ros with something like what you can do with topics.  I couldn't find any documentation on that.

```suggestion
  auto const temp_node = std::make_shared<rclcpp::Node>("synchronized_string_parameter");
  string_subscriber_ = temp_node->create_subscription<std_msgs::msg::String>(
```

Reply:
ROS 1 had a way to create an "anonymous node," but I don't see a way to do it in ROS 2.  But I changed the name to something more unique.